atlasdb-dbkvs/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/OracleTableNameGetterTest.java

```diff
@@ -0,0 +1,178 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.dbkvs;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableSet;
+import com.palantir.atlasdb.keyvalue.api.Namespace;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.ConnectionSupplier;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.OverflowMigrationState;
+import com.palantir.common.exception.TableMappingNotFoundException;
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class OracleTableNameGetterTest {
+    private static final OracleDdlConfig TABLE_MAPPING_DDL_CONFIG = ImmutableOracleDdlConfig.builder()
+            .overflowMigrationState(OverflowMigrationState.UNSTARTED)
+            .useTableMapping(true)
+            .build();
+
+    private static final OracleDdlConfig NON_TABLE_MAPPING_DDL_CONFIG = ImmutableOracleDdlConfig.builder()
+            .overflowMigrationState(OverflowMigrationState.UNSTARTED)
+            .useTableMapping(false)
+            .build();
+
+    private static final Set<String> SHORT_TABLE_NAMES = ImmutableSet.of("shortNameOne", "shortNameTwo");
+    private static final Set<TableReference> TABLE_REFERENCES = ImmutableSet.of(
+            TableReference.create(Namespace.create("test"), "world"), TableReference.createWithEmptyNamespace("hello"));
+
+    @Mock
+    private ConnectionSupplier connectionSupplier;
+
+    @Mock
+    private OracleTableNameMapper tableNameMapper;
+
+    @Mock
+    private OracleTableNameUnmapper tableNameUnmapper;
+
+    private OracleTableNameGetter tableMappingTableNameGetter;
+    private OracleTableNameGetter nonTableMappingTableNameGetter;
+
+    @Before
+    public void before() {
+        tableMappingTableNameGetter =
+                new OracleTableNameGetter(TABLE_MAPPING_DDL_CONFIG, tableNameMapper, tableNameUnmapper);
+        nonTableMappingTableNameGetter =
+                new OracleTableNameGetter(NON_TABLE_MAPPING_DDL_CONFIG, tableNameMapper, tableNameUnmapper);
+    }
+
+    @Test
+    public void getTableReferencesFromShortTableNamesTransformsUnmapperNamesWhenMappingEnabled()
```

Comment:
These tests look parameterisable (given they are various combinations of table mapping, non table mapping e.t.c), but parameterising (to me) ended up less readable given all the combinations